From e52628de35faf23cf9c944070808c551aa4134ca Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 13 Nov 2023 16:15:14 +0200
Subject: [PATCH] [SW-66327] pif_arc: gaudi3: clear pci termination

In secure boot mode, the access from PCIe to NoC is blocked by HW FSM.
Due to this, any access from PCIe I/F to ASIC will be blocked.
FW need to clear boot termination in order to un-block the access.
According to embedded PRM 3.8.3.2.1 the termination should also be
cleared during PCIe init.

Change-Id: Icb1d7d5aa3b9528784ac59558372087c7423a87b
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index c395f0a..7a76a57 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1083,6 +1083,9 @@
 	/* Wait for phy fw upgrade completion */
 	pcie_wait_for_sram_ext_ld_done();
 
+	/* clear boot termination */
+	hl_pcie_clear_boot_termination();
+
 	/* Enable and Mask LTSSM */
 	regmap_writel(LTTSM_ENA_MASK, D0_PCIE_AUX_REG(mmPCIE_AUX_LTSSM_EN));
 }
diff --git a/arch/gaudi3/arc/hl.h b/arch/gaudi3/arc/hl.h
index f3fbe40..e676700 100644
--- a/arch/gaudi3/arc/hl.h
+++ b/arch/gaudi3/arc/hl.h
@@ -51,5 +51,6 @@
 u32 gaudi3_die_timestamp_base_get(u32 die_id);
 int gaudi3_hbw_memcpy(u64 src, u64 dst, u32 size);
 void pif_arc_init_pcie(void);
+void hl_pcie_clear_boot_termination(void);
 
 #endif /* ARCH_GAUDI3_ARC_HL_H_ */
